### ISSUE TITLE
Feature: Built-in support for IRC in Nyxt?

### ISSUE THREAD

**Issue by guojing0** [atlas-engineer/nyxt#1937]
I was reading Demeter articles on official blog and then it came to me that Nyxt seemed to not have support for IRC (at least it's the result when I searched in GitHub)? After some research I found out that maybe [maiden](https://shirakumo.github.io/maiden/) and [cl-irc](https://www.common-lisp.net/project/cl-irc/) can be helpful.

I hope to implement this after completing most items in my to-do list... either as a library/mode or a plug-in (like Demeter).

**Comment by hendursaga** [comment]
@guojing0, you might want to check out [Mezzano](https://mezzanos.herokuapp.com/) the Common Lisp OS, they have an [IRC app](https://github.com/froggey/Mezzano/blob/f0eeb2a3f032098b394e31e3dfd32800f8a51122/applications/irc.lisp) you could draw inspiration/code from.

**Comment by jmercouris** [comment]
I would also like to do this, if you get started on this, please do let me know so that we are not doing the same work twice :-)

**Comment by guojing0** [comment]
@hendursaga Thanks. I didn't know Mezzano also has an IRC app.

**Comment by guojing0** [comment]
@jmercouris Of course. I was thinking about doing this after finishing that button thing and "headings buffer". Maybe I could start work on this during the break.

**Comment by jmercouris** [comment]
OK! :-)

**Comment by guojing0** [comment]
### UX/UI design of IRC interface

As we can see from the following screenshot, Demeter has three main parts (list of feeds, specific feed, and the article) with prompt buffer. I am thinking about implementing something like [Textual](https://www.codeux.com/textual/)'s GUI, which means that for the prompt buffer, we could only need the prompt without the area below it (if it can be done).

Any thoughts? @Ambrevar @jmercouris @aartaka @pdelfino 

![demeter](https://user-images.githubusercontent.com/3425792/146678467-c944af2e-d2e8-4b13-911a-8f0b00339402.png)

**Comment by aartaka** [comment]
> As we can see from the following screenshot, Demeter has three main parts (list of feeds, specific feed, and the article) with prompt buffer. I am thinking about implementing something like [Textual](https://www.codeux.com/textual/)'s GUI, which means that for the prompt buffer, we could only need the prompt without the area below it (if it can be done).

Doesn't sound right to me. The area below prompt input is useful, as it allows listing arbitrary suggestions information.

But, if you insist, prompt-buffer shrinking is definitely possible. `ffi-window-set-prompt-buffer-height` can be of help there -- just set it to input height and you're all set for the suggestion-less prompt!

**Comment by guojing0** [comment]
May I ask how IRC support is for Nyxt now?

**Comment by aartaka** [comment]
It didn't change much since the last discussion :P
